### ISSUE TITLE
Issue with k8s.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/

### ISSUE THREAD

**Issue by johnstosh** [kubernetes/website#15048]
Very nice help page. This was instrumental in my moving forward.

Could you enhance a small part of this help page?

**Problem:**
The text:  "For a probe, the kubelet makes the probe connection at the node, not in the pod, which means that you can not use a service name in the host parameter since the kubelet is unable to resolve it."  almost makes sense, but not quite.  Can you provide a URL to another page or explain in more detail?  Is the probe executed inside the container?

**Proposed Solution:**
Please explain in more detail the context of execution for the probe and the evaluation of the parameters.

**Page to Update:**
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/

**Comment by sftim** [comment]
That's a good suggestion @johnstosh 

kubelet makes the connection for a probe. When kubelet resolves hostnames, it's relying on the node's stub resolver and DNS services, and it's not querying cluster DNS (which is what Pods and their containers do).
Not really sure that the wording I've put there is any better though!

/language en
/kind feature

**Comment by Coderhypo** [comment]
Hello, 

I guess you are wondering about how kubelet request the web server in pod(if use http probe) when kubelet do probe.

and I looked up the source code to describe the process.

## How kubelet start a http probe：

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/prober.go#L162

if we use http probe，kubelet will get `scheme `、`host`、`port`、`path`，Thus obtaining a complete address（if you not define `host`，kubelet will use podIP）

## How kubelet request

https://github.com/kubernetes/kubernetes/blob/master/pkg/probe/http/http.go#L68

kubelet will create a **normal** http client(go package `net/http`) when get the url，and do request.

to sum up，**The web request is initiated by the kubelet process **on node**, not the process inside the container.**

Have I solved your doubts?

**Comment by johnstosh** [comment]
Hi guys,

Thanks for the replies. This is very helpful.

So, to summarize: The probe is executed by the kubelet inside the node, but outside the container. And the probe parameters/values go straight from the manifest to the kubelet without being evaluated or converted by a global piece of code that knows all the mnemonics and symbols. 

For example, the host name: When kubelet resolves hostnames, it's relying on the node's stub resolver and DNS services. This explains why the service name can't be used for the host because it would have to be resolved to an IP address by the cluster DNS.

Thanks !

Kind Regards,
Johnny
